### PR TITLE
Experimental speech streaming for LMNT (useChat/useCompletion React)

### PR DESCRIPTION
**Pull request by lgrammel**
## Summary

Adds speech streaming to `useChat` and `useCompletion` with `streamData`.

- `useCompletion` & `useChat` (for React) provide a `experimental_speechUrl` that can be used html `audio` elements
- Integration functions for lmnt speech streams through `experimental_forwardLmntSpeechStream`
- `streamData.experimental_appendSpeech`: add speech stream chunks to data stream (used automatically through forward functions)
- Example: `examples/next-lmnt`: LMNT completion & chat speech streaming
- Docs: LMNT provider docs, API docs for `experimental_forwardLmntSpeechStream`

## Notes
- The LMNT SDK does not work in the edge environment (as of `v1.1.2`)

### CODE REVIEW

**Comment by untilhamza** [comment]
This is exciting

**Comment by llermaly** [comment]
This is awesome!

**Comment by llermaly** [comment]
@lgrammel Hi Lars, I tried to test this one locally with no luck, it is showing this error: 

```
 ⚠ ./app/api/chat-speech-elevenlabs/route.ts
Attempted import error: 'forwardModelFusionSpeechStream' is not exported from 'ai' (imported as 'forwardModelFusionSpeechStream').
```
I go to node_modules/ai and I see the function there, not sure if I need to do anything else. (I cloned the fork, checkout to the branch and run the example)

It is ready to test? 

Thanks!

**Comment by lgrammel** [comment]
> @lgrammel Hi Lars, I tried to test this one locally with no luck, it is showing this error:
> 
> ```
>  ⚠ ./app/api/chat-speech-elevenlabs/route.ts
> Attempted import error: 'forwardModelFusionSpeechStream' is not exported from 'ai' (imported as 'forwardModelFusionSpeechStream').
> ```
> 
> I go to node_modules/ai and I see the function there, not sure if I need to do anything else. (I cloned the fork, checkout to the branch and run the example)
> 
> It is ready to test?
> 
> Thanks!

Have you rebuilt the ai package? The easiest way is to just rebuild the whole repository (`pnpm i`, `pnpm build`) and then try out the example.

**Comment by llermaly** [comment]
> > @lgrammel Hi Lars, I tried to test this one locally with no luck, it is showing this error:
> > ```
> >  ⚠ ./app/api/chat-speech-elevenlabs/route.ts
> > Attempted import error: 'forwardModelFusionSpeechStream' is not exported from 'ai' (imported as 'forwardModelFusionSpeechStream').
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I go to node_modules/ai and I see the function there, not sure if I need to do anything else. (I cloned the fork, checkout to the branch and run the example)
> > It is ready to test?
> > Thanks!
> 
> Have you rebuilt the ai package? The easiest way is to just rebuild the whole repository (`pnpm i`, `pnpm build`) and then try out the example.

That did the trick. I was doing `npm run dev` , I did `pnpm build` , `npm start` and it worked.

I works really, really fast. I hope we can get this merged very soon.

**Comment by llermaly** [comment]
Hi @MaxLeiter! did you have a chance to take a look?

**Comment by tgonzales** [comment]
Hello @lgrammel I saw that you changed from eleven labs to LMNT, there is a technical reason for this, eleven labs supports multi languages, LMNT still has no plans to launch this, wouldn't it be interesting to keep both options?

Thank you and congratulations for the excellent work

**Comment by Iven2132** [comment]
@lgrammel Hi! I can't find the example app for speech streaming in the Vercel AI SDK repo. where it's gone?

**Comment by lgrammel** [comment]
> @lgrammel Hi! I can't find the example app for speech streaming in the Vercel AI SDK repo. where it's gone?

this feature has not been merged yet

**Comment by Iven2132** [comment]
> > @lgrammel Hi! I can't find the example app for speech streaming in the Vercel AI SDK repo. where it's gone?
> 
> this feature has not been merged yet

Yeah, But i want to use it, is there any way i can use it?

**Comment by Iven2132** [comment]
Hi @MaxLeiter Can you merge this?

**Comment by Iven2132** [comment]
> > @lgrammel Hi! I can't find the example app for speech streaming in the Vercel AI SDK repo. where it's gone?
> 
> this feature has not been merged yet

When this feature will be merged?

**Comment by Iven2132** [comment]
@llermaly How did you use this?

**Comment by Iven2132** [comment]
Hi @MaxLeiter Can you please approve this?

**Comment by llermaly** [comment]
bump

**Comment by pixelcatgg** [comment]
we could really use this as well 🙏 thank you so much for the work on this

File: examples/next-lmnt/app/api/chat/route.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// Create an OpenAI API client
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY || '',
+});
+
+// Create an LMNT API client
+const speech = new Speech(process.env.LMNT_API_KEY || 'no key');
+
+// Note: The LMNT SDK does not work on edge yet (as of v1.1.2)
+// export const runtime = 'edge';
```
Comment (shaper) [inline examples/next-lmnt/app/api/chat/route.ts:19]: Hi @lgrammel FYI @kaikato just merged a short README in `lmnt-node` describing how we got this working -- if you see an even better way let us know, but with the one change to the `next.config.js` file it should work with `edge`. https://github.com/lmnt-com/lmnt-node/pull/32

Comment (shaper) [reply examples/next-lmnt/app/api/chat/route.ts:19]: I thought to add that when we were hacking on https://github.com/vercel/ai-chatbot/pull/151 I did see issues that looked like a challenge re: websockets staying alive on `edge` and so for deployment I switched to `nodejs` and didn't look further at the time. You can see the deployment focused work I did atop that PR here: https://github.com/shaper/lmnt-ai-chatbot/commits/main/

Comment (Iven2132) [reply examples/next-lmnt/app/api/chat/route.ts:19]: @lgrammel @MaxLeiter When TTS will come?

**Comment by allenchuang** [comment]
Any reason why this is closed? TTS is a great feature to have

**Comment by solanacryptodev** [comment]
Would be cool to see TTS added with the addition of gpt-4o

**Comment by alokwhitewolf** [comment]
@lgrammel / @MaxLeiter 
Any follow up plans on adding TTS to vercel AI ?

**Comment by alicercedigital** [comment]
we are very excited about this!

**Pull request closed without merging**